storage/cleanup_test.go

```diff
@@ -117,6 +113,111 @@ func TestCleanupManagerCommitLogTimeRange(t *testing.T) {
 	require.Equal(t, time.Unix(7200, 0), ce)
 }
 
+func newTestRetentionOptions(
+	blockSizeSecs int,
+	bufferPastSecs int,
+	bufferFutureSecs int,
+) retention.Options {
+	return retention.NewOptions().
+		SetBufferPast(time.Duration(bufferPastSecs) * time.Second).
+		SetBufferFuture(time.Duration(bufferFutureSecs) * time.Second).
+		SetBlockSize(time.Duration(blockSizeSecs) * time.Second)
+}
+
+type testCaseCleanupMgrNsBlocks struct {
+	// input
+	id          string
+	nsRetention retention.Options
+	clBlockSize time.Duration
+	blockStart  time.Time
+
+	// output
+	expectedStart time.Time
+	expectedEnd   time.Time
+}
+
+func TestCleanupManagerCommitLogNamespaceBlocks(t *testing.T) {
+	tcs := []testCaseCleanupMgrNsBlocks{
+		{
+			id:            "test-case-0",
+			nsRetention:   newTestRetentionOptions(30, 0, 0),
```

Comment:
nit: Can you perhaps make this a struct options to retention options?  It's hard to read these tests.

Whereas `newTestRetentionOptions(testRetentionOptions{blockSizeSecs: 30, bufferPastSecs: 0, bufferFutureSecs: 0})` is a lot more readable.